src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1360,64 +1360,56 @@ void Instance::OnFailSafeTimerExpired()
     }
 }
 
-CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster,
+                                               DataModel::ListBuilder<DataModel::AcceptedCommandEntry> & builder)
 {
     using namespace Clusters::NetworkCommissioning::Commands;
+    using Priv = chip::Access::Privilege;
 
-    if (mFeatureFlags.Has(Feature::kThreadNetworkInterface))
-    {
-        for (auto && cmd : {
-                 ScanNetworks::Id,
-                 AddOrUpdateThreadNetwork::Id,
-                 RemoveNetwork::Id,
-                 ConnectNetwork::Id,
-                 ReorderNetwork::Id,
-             })
-        {
-            VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
-        }
-    }
-    else if (mFeatureFlags.Has(Feature::kWiFiNetworkInterface))
+    static constexpr size_t kNetworkCommands = 7; // Count of max possible commands assuming all features
+    ReturnErrorOnFailure(builder.EnsureAppendCapacity(kNetworkCommands));
+
+    bool hasThread = mFeatureFlags.Has(Feature::kThreadNetworkInterface);
+    bool hasWifi   = mFeatureFlags.Has(Feature::kWiFiNetworkInterface);
+
+    if (hasThread | hasWifi)
     {
-        for (auto && cmd : {
-                 ScanNetworks::Id,
-                 AddOrUpdateWiFiNetwork::Id,
-                 RemoveNetwork::Id,
-                 ConnectNetwork::Id,
-                 ReorderNetwork::Id,
-             })
-        {
-            VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
-        }
+        ReturnErrorOnFailure(builder.AppendElements({
+            { ScanNetworks::Id, {}, Priv::kAdminister },   //
```

Comment:
The fact that all this code has to duplicate parts of the spec is really fragile.

What we should be doing is that cluster-objects codegen should be code-generating the relevant bits: either a getter for the `AcceptedCommandEntry` or at least the privileges/flags bits that can then be used to construct one.

I'm OK with this happening in a followup that then updates all these callsites, or in a PR that happens before this PR, either way.  Just not as part of this PR.

But we don't want to be shipping this in the state this PR is creating, where all sorts of spec stuff is duplicated in the cluster impls.

Changes requested until we have a concrete plan for how this is going to work.

Reply:
Some of the logic issues may be pre-existing as we still controlled IDs within code.
Is the worry that we may make mistakes in setting correct flags/privileges for these items?

i.e. are you asking for something like `ScanNetworks::Entry` to be generated by cluster-objects so that we do not make mistakes in flags and privileges?

Reply:
>  Is the worry that we may make mistakes in setting correct flags/privileges for these items?

Yes, 100%.  As this PR demonstrates, in several places.  ;)

> i.e. are you asking for something like ScanNetworks::Entry to be generated by cluster-objects so that we do not make mistakes in flags and privileges?

Yes.